span-normalizer/raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/span_attribute.proto

```diff
@@ -174,5 +174,5 @@ enum TracerAttribute {
 
 enum JaegerAttribute {
     JAEGER_ATTRIBUTE_UNSPECIFIED = 0 [(string_value) = "unspecified"];
-    JAEGER_ATTRIBUTE_SERVICE_NAME = 1 [(string_value) = "jaeger.servicename"];
+    JAEGER_ATTRIBUTE_SERVICE_NAME = 1 [(string_value) = "servicename"];
```

Comment:
I think, need to check but the agent can still send data in this `jaeger.servicename`. So, we need to parse from this and store it in the service name. Let me check.

Reply:
I think this is an important, but this logic belong to place where we are converting to Jaeger format in collector. 
Ideally we would assume for the following code block 
```if (jaegerSpan.getProcess() != null
        && !StringUtils.isEmpty(jaegerSpan.getProcess().getServiceName())) {
      // Keep the attribute for now due to backwards compatibility
      attributeFieldMap.put(
          RawSpanConstants.getValue(JaegerAttribute.JAEGER_ATTRIBUTE_SERVICE_NAME),
          AttributeValueCreator.create(jaegerSpan.getProcess().getServiceName()));
      eventBuilder.setServiceName(jaegerSpan.getProcess().getServiceName());
    }```
jaegerSpan.getProcess().getServiceName() is populated in the upstream. 
I am not sure if that logic is already there in the collector cc @pavolloffay  @jcchavezs

Reply:
As this constant is part of API version v1, I would suggest not change the value of it, and as our data model contains service_name (https://github.com/hypertrace/data-model/blob/main/data-model/src/main/avro/Event.avdl), let's consume from it everywhere. Once those changes are merged,
we can just remove populating this field, and if requires, we can remove constants - JaegerAttribute{} - marking as depercated.

Reply:
Any application reporting jaeger spans will have this attribute. We are kind of stuck with this till the end of times.

Reply:
I am not sure I am following, can't we accept both and then override serviceName if it is empty with jaegerServiceName in the entire pipeline?

Reply:
I agree, I am now looking for service name in either jaegerSpan.getProcess().getServiceName() or against the key `jaeger.servicename` in the attributemap.

Reply:
Though I am not passing the old attribute key `jaeger.servicename` downstream, since it shows up in the ui under attribute list.

Reply:
The collector (OT/OTEL) might add jaeger service name as tag in the internal conversion in Jaeger receiver. Then in the exporter (e.g. kafka) the model is again converted from internal model(x) to Jaeger and the tag might not be removed. Our platform could handle this data duplication.

(x) OTEL collector uses internally OTEL span, I think the OC collector  might do something similar.

Reply:
> Then in the exporter (e.g. kafka) the model is again converted from internal model(x) to Jaeger and the tag might not be removed.

I think we aren't affected by this as whatever comes we do the fix in the pipeline. The main goal here is to get rid of the attribute and populate serviceName.

Maybe a quick branch on top of your integration test might show this @rish691

Reply:
Yes, we will be able to populate the service name correctly even if its sent against the key `jaeger.servicename`. 

@jcchavezs  adding a test on top of the integration test won't work (as of now), as trace-enricher needs to connect to entity-service for packet processing(that is why we have a very simple packet in e2e test). 

Though I think the existing test in this pr should suffice.

Reply:
@rish691 Let's not change the value in proto as recommended. Internally,  we read it and populate in service name which you have already done, no need to introduce one more constant `OLD_JAEGER_SERVICENAME_KEY`